src/core/batch.rkt

```diff
@@ -145,6 +153,86 @@
       [_ node]))
   (unmunge reg))
 
+(define (batch-restore-index batch)
+  (make-hash (for/list ([node (in-vector (batch-nodes batch))]
+                        [n (in-naturals)])
+               (cons node n))))
+
+(define (egg-nodes->batch egg-nodes id->spec input-batch rename-dict)
+  (define out (make-mutable-batch))
+  (set-mutable-batch-index! out (batch-restore-index input-batch))
```

Comment:
Ah I see. Can we wrap this in `batch->mutable-batch`?

Reply:
Or even just require that the input is mutable? Wait, isn't it already mutable?

Reply:
Yeah, we can make this function, at least we have 2 occurrences of this code.
And answering the question, this `input-batch` is immutable, it works like an aggregator.
Maybe it will make sense to keep `input-batch` mutable at the rewriting stage, and once we are done with `rr` and `taylor` - make it immutable.